python/graphstorm/gconstruct/utils.py

```diff
@@ -997,12 +1005,33 @@ def partition_graph(g, node_data, edge_data, graph_name, num_partitions, output_
                  num_test != g.number_of_nodes(ntype)):
             balance_arr -= 1
         balance_ntypes[ntype] = balance_arr
-    mapping = \
-        dgl.distributed.partition_graph(g, graph_name, num_partitions, output_dir,
-                                        part_method=part_method,
-                                        balance_ntypes=balance_ntypes,
-                                        balance_edges=True,
-                                        return_mapping=save_mapping)
+
+    # Call DGL's partition graph, using the appropriate arg list
+    # for each version
+    dgl_version_str = importlib.metadata.version("dgl")
+    dgl_version = version.parse(dgl_version_str)
+    partition_kwargs = {
+        "g": g,
+        "graph_name": graph_name,
+        "num_parts": num_partitions,
+        "out_path": output_dir,
+        'part_method': part_method,
+        'balance_ntypes': balance_ntypes,
+        'balance_edges': True,
+        'return_mapping': save_mapping,
+    }
+
+    if dgl_version >= version.parse("2.0.0"):
```

Comment:
Do we need to check for 2.3.0+ or just 2.0.0?

Reply:
Good catch, seems like the new argument was introduced in 2.1.0. I'll update the checks accordingly